src/etc/inc/services.inc

```diff
@@ -1748,7 +1748,7 @@ EOD;
                 $need_update = false;
 
                 /* Update IPv4 if we have it. */
-                if (is_ipaddrv4($wanip)) {
+                if (is_ipaddrv4($wanip) && $dnsupdate['recordtype'] != "AAAA") {
```

Comment:
logic works, but may not be resilient against broadening recordtype to other types, maybe:

is_ipaddrv4($wanip) && (empty($dnsupdate['recordtype']) || $dnsupdate['recordtype'] == 'A')